### ISSUE TITLE
BulkInsertOrUpdate temp tables are not deleted on Npgsql.PostgresException

### ISSUE THREAD

**Issue by yhan** [borisdj/EFCore.BulkExtensions#1030]
Hi,

My BulkInsertOrUpdate operations are executed in a single thread.

`BulkConfig::UseTempDB` is not used ( false).

The code running into error is simply
DbContext.BulkInsertOrUpdate(IListw<T> ...) - without `BulkConfig`.

Upon following exception, temp table `OrdersTemp7a07a620` is not deleted:
The problem is produced from time to time: ~40 failures per day. The process inserts every 1s during 9 hours per day.

```
|2022-12-16 09:55:02.778322 [Ingestion] ERROR [Microsoft.EntityFrameworkCore.Database.Command] Failed executing DbCommand (0ms) [Parameters=[], CommandType='Text', CommandTimeout='30']

DROP TABLE IF EXISTS "**OrdersTemp7a07a620**"

|2022-12-16 09:55:02.779213 [Ingestion] ERROR [<obfuscated>.Persistence.Repository.WriteRepository] flush 76 <obfuscated>.Persistence.DbEntity.Order failed, wait for next round

Npgsql.PostgresException (0x80004005): XX000: tuple concurrently updated

   at Npgsql.Internal.NpgsqlConnector.<ReadMessage>g__ReadMessageLong|211_0(NpgsqlConnector connector, Boolean async, DataRowLoadingMode dataRowLoadingMode, Boolean readingNotifications, Boolean isReadingPrependedMessage)

   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming, CancellationToken cancellationToken)

   at Npgsql.NpgsqlDataReader.NextResult()

   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)

   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)

   at Npgsql.NpgsqlCommand.ExecuteNonQuery(Boolean async, CancellationToken cancellationToken)

   at Npgsql.NpgsqlCommand.ExecuteNonQuery()

   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteNonQuery(RelationalCommandParameterObject parameterObject)

   at Microsoft.EntityFrameworkCore.RelationalDatabaseFacadeExtensions.ExecuteSqlRaw(DatabaseFacade databaseFacade, String sql, IEnumerable`1 parameters)

   at EFCore.BulkExtensions.SQLAdapters.PostgreSql.PostgreSqlAdapter.MergeAsync[T](DbContext context, Type type, IList`1 entities, TableInfo tableInfo, OperationType operationType, Action`1 progress, CancellationToken cancellationToken, Boolean isAsync)

   at EFCore.BulkExtensions.SQLAdapters.PostgreSql.PostgreSqlAdapter.Merge[T](DbContext context, Type type, IList`1 entities, TableInfo tableInfo, OperationType operationType, Action`1 progress)

   at EFCore.BulkExtensions.DbContextBulkTransaction.Execute[T](DbContext context, Type type, IList`1 entities, OperationType operationType, BulkConfig bulkConfig, Action`1 progress)

   at EFCore.BulkExtensions.DbContextBulkExtensions.BulkInsertOrUpdate[T](DbContext context, IList`1 entities, BulkConfig bulkConfig, Action`1 progress, Type type)

   at <obfuscated>.Persistence.Repository.WriteRepository`2.Save(List`1 entities) in <obfuscated-path>\<obfuscated>\<obfuscated>.Persistence.Bulk\Repository\I<obfuscated>WriteRepository.cs:line 49

  Exception data:

    Severity: ERROR

    SqlState: XX000

    MessageText: tuple concurrently updated

    File: heapam.c

    Line: 2884

    Routine: simple_heap_delete

```

Referenced packages:
```
<PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="6.0.3" />
<PackageReference Include="EFCore.BulkExtensions" Version="6.4.0" />
```

Can you please suggest?
Many thanks!

Yi

**Comment by yhan** [comment]
> above tested both in postgres v12 and v14

tried also using temp table:

```csharp
 using var context = dbctxFactory.CreateDbContext();
 using var transaction = context.Database.BeginTransaction(); 
 var bulkConfig = new BulkConfig 
 { 
     //PreserveInsertOrder = true, // true is default 
     SetOutputIdentity = true, 
     BatchSize = 4000, 
     UseTempDB = true, 
     CalculateStats = true 
 }; 
 
 context.BulkInsert(entities, bulkConfig); 
 transaction.Commit();
```

all temp tables stay - non of them are dropped.

I might probably something here ...

**Comment by yhan** [comment]
with `UseTempDB=true`.

Under below temp  tables are not dropped:
 ```
<PackageReference Include="EFCore.BulkExtensions" Version="6.6.2" />
        <PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="7.0.0" />
        <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="6.0.11" />
```

with an older version temp tables are correctly dropped:

```
        <PackageReference Include="EFCore.BulkExtensions" Version="6.5.6" />
        <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="7.0.0" />
```

code attached:
[BulkExtensionsTempTableNotDropped.zip](https://github.com/borisdj/EFCore.BulkExtensions/files/10253694/BulkExtensionsTempTableNotDropped.zip)

**Comment by borisdj** [comment]
Should be fixed with 7.0.4
